lib/wickedbase.pm

```diff
@@ -858,17 +858,22 @@ sub skip_check_logs_on_post_run {
     shift->{skip_check_logs_on_post_run} = 1;
 }
 
+sub check_logs_cursor {
+    my $cursor = script_output(q(journalctl -o export -n 1 | tr -dc '\n|[[:print:]]' |  grep __CURSOR));
+    ($cursor) = ($cursor =~ /^__CURSOR=(.*)$/m);
+    return $cursor;
+}
+
 sub check_logs {
     my $self = shift;
     my $code = shift;
-    my $cursor = '';
+    my $cursor = $self->{pre_run_check_logs_cursor} // '';
 
     if (ref($code) eq 'CODE') {
-        $cursor = script_output(q(journalctl -o export -n 1 | tr -dc '\n|[[:print:]]' |  grep __CURSOR));
-        ($cursor) = ($cursor =~ /^__CURSOR=(.*)$/m);
-        $cursor = "-c '$cursor'";
+        $cursor = $self->check_logs_cursor();
```

Comment:
can you elaborate why we need to redefine cursor second time in this case ?

Reply:
and why we can not just do here : 
```suggestion
        $self->{pre_run_check_logs_cursor} = $self->check_logs_cursor();
```

this would made all code a bit clearer

Reply:
It's for calls like this: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/wickedbase.pm#L838
which is used here https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/wicked/startandstop/sut/t17_ifrename.pm#L24 and what we wanna get a log error specific for each `run_test_shell_script()` call, so we can actually map on which call the error occurred.

Reply:
Maybe I misunderstood you. It would work to set the `$self->{pre_run_log_cursor}` variable direclty. But then we would have two places where it is set, which makes the code harder to read and we would loose the functionality where we wanna check logs specific on  a call **and** at the end for the full runtime of the test-module.

Reply:
If we would do it, then I would actually also set the `skip_check_logs_on_post_run()` if `check_logs()` is called with `CODE`. So we actually forbid the checking of full test-module logs once someone call `run_test_shell_script()`.

Reply:
ehm maybe I am wrong but `__CURSOR=` is set only once , no ? `get_log_cursor` is just getting it from journal so no matter *when * you will call `get_log_cursor` you will always get same place

Reply:
no, its like a timestamp and if you pass this to a subsequent `journalctl` call you will get all message from that "timestamp" to `now`.